plugin/pkg/admission/taint/admission.go

```diff
@@ -0,0 +1,282 @@
+	return c.client.ClusterRoleBindings().List(*options)
+}
+
+func (c *rbacClient) GetRole(ctx api.Context, id string) (*rbac.Role, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	role, err := c.client.Roles(ns).Get(id)
+	if err != nil {
+		return nil, fmt.Errorf("get role: %v", err)
+	}
+	return role, nil
+}
+
+func (c *rbacClient) ListRoleBindings(ctx api.Context, options *api.ListOptions) (*rbac.RoleBindingList, error) {
+	ns, ok := api.NamespaceFrom(ctx)
+	if !ok {
+		return nil, errors.New("no namespace found")
+	}
+	roleBindings, err := c.client.RoleBindings(ns).List(*options)
+	if err != nil {
+		return nil, fmt.Errorf("list role bindings: %v", err)
+	}
+	return roleBindings, nil
+}
+
+func init() {
+	admission.RegisterPlugin("TaintAdmit", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return NewTaintAdmit(client, config), nil
+	})
+}
+
+// TaintAdmit is an implementation of admission.Interface which taints nodes at creation time
+type taintAdmit struct {
+	client clientset.Interface
+}
+
+// Determine whether a node is trusted
+func isTrusted(node *api.Node) (bool, error) {
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	if err != nil {
+		return true, err
+	}
+	for _, taint := range taints {
+		if taint.Key == TaintKey {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+func toleratesUntrusted(pod *api.Pod) (bool, error) {
+	tolerations, err := api.GetTolerationsFromPodAnnotations(pod.Annotations)
+	if err != nil {
+		return false, err
+	}
+	for _, toleration := range tolerations {
+		if toleration.Key == TaintKey {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// Flag a node as untrusted
+func invalidateNode(node *api.Node) error {
+
+	if node.Annotations == nil {
+		node.Annotations = make(map[string]string)
+	}
+	taints, err := api.GetTaintsFromNodeAnnotations(node.Annotations)
+	newTaints := []api.Taint{}
+	untrustedTaint := api.Taint{
+		Key:    TaintKey,
+		Value:  "true",
+		Effect: api.TaintEffectNoSchedule,
+	}
+	for _, taint := range taints {
+		if taint.Key == TaintKey {
+			continue
+		}
+		newTaints = append(newTaints, taint)
+	}
+	newTaints = append(newTaints, untrustedTaint)
+	jsonContent, err := json.Marshal(newTaints)
+	if err != nil {
+		glog.Errorf("Unable to marshal new taints for %s", node.GetName())
+	}
+	node.Annotations[api.TaintsAnnotationKey] = string(jsonContent)
+	return err
+}
+
+func (t *taintAdmit) Admit(a admission.Attributes) (err error) {
+	kind := a.GetKind().GroupKind()
+
+	user := a.GetUserInfo()
+	// Allow access over the insecure socket to do anything
+	if user == nil {
+		return nil
+	}
+
+	if kind != api.Kind("Node") && kind != api.Kind("ConfigMap") && kind != api.Kind("Pod") {
+		return nil
+	}
+
+	configmap, err := t.client.Core().ConfigMaps("default").Get("taint.coreos.com")
```

Comment:
This also feels like something that should be cached, rather than retrieved on each request.